src/main/java/com/pss/ProjectileSimulationSuite.java

```diff
@@ -5,59 +5,86 @@
 import com.pss.handlers.*;
 
 import javax.vecmath.Vector3d;
+import java.util.ArrayList;
+import java.util.List;
 
+/**
+ * This class provides a suite for running projectile simulations.
+ * The simulations steps are computed, stored, and output to various formats.
+ */
 public class ProjectileSimulationSuite {
-    private static Vector3d[] _results;
+
+    private static int MAX_SIMSTEPS = 100;
+    private static int steps;
     private static IOutputResults[] _resultsOutputers;
-    private static final int MAX_OUTPUTERS = 4;
-    // private static final int MAX_OUTPUTERS = 3;
-    private static final int MAX_SIMSTEPS = 80;
-    private static IGetConfiguration _configHandler;
     private static ProjectileSimulator _simulator;
+    private static List<Vector3d> _results;
+    private static Vector3d[] _sim_steps;
 
+    /**
+     * The main method of the suite. It initializes the simulation, runs it,
+     * and then outputs the results.
+     *
+     * @param args Command line arguments
+     */
     public static void main(String[] args) {
         initSimulation();
 
         for (int t = 0; t < MAX_SIMSTEPS; t++) {
-            _results[t] = new Vector3d(0, 0, 0);
-            _results[t].add(_simulator.updatePosition());
+            _sim_steps[t] = new Vector3d(0, 0, 0);
+            _sim_steps[t].add(_simulator.updatePosition());
+            // Increment steps counter if the z coordinate is non-negative
+            if (_sim_steps[t].z >= 0)
+                steps++;
+        }
+
+        // Store the results
+        _results = new ArrayList<>(steps);
+        for (int i = 0; i < steps; i++) {
+            _results.add(_sim_steps[i]);
         }
-        
+
         outputResults();
     }
 
+    /**
+     * Initializes the simulation by getting outputers and creating a new simulator.
+     * The precision of the simulator is set by dividing the maximum number of
+     * simulation steps set by the precision of the time steps used in computation.
+     *
+     * The default setting is 100 steps at 0.001 precision which results in 100,000
+     * simulation steps total.
+     */
     private static void initSimulation() {
-        _results = new Vector3d[MAX_SIMSTEPS];
         _resultsOutputers = getOutputers();
-        _configHandler = new FileGetConfiguration();
-
-        _simulator = new MakeProjectileSimulator().createProjectileSimulator(_configHandler);
-    };
+        _simulator = new MakeProjectileSimulator().createProjectileSimulator(new FileGetConfiguration());
```

Comment:
Java moment: the result of a make function calls a create function. Don't need to change, just wanted to point it out hahaha